### PR TITLE
Decompile sub_65E1C2

### PR DESCRIPTION
**Pull request by kevinburke**

### CODE REVIEW

File: src/ride/ride_ratings.c [hunk 1]
```diff
@@ -619,11 +619,35 @@ static rating_tuple sub_65DDD1(rct_ride *ride)
...
+	eax = min(var_118_shifted, 2000);
+	int intensity = (eax * 0x2666) >> 16;
+
+	eax = min(var_118_shifted, 1000);
+	int nausea = (eax * 0x4000) >> 16;
+
+	eax = (ride->var_11C * 30340) >> 16;
+	nausea += eax;
+
+	if (ride->var_11E & 0x40) {
```
Comment (kevinburke) [inline src/ride/ride_ratings.c]: Should probably turn these into flags

**Comment by duncanspumpkin** [comment]
From looking at how these are set var_118 is the distance under cover. var_11E bits 0x1F are how many times it goes undercover. Unsure on the flags or anything to do with var_11C. I think we can probably rename the function undercover_rating or something similar.

**Comment by kevinburke** [comment]
maybe underground instead of undercover?


--
Kevin Burke
phone: 925.271.7005 | twentymilliseconds.com

**Comment by duncanspumpkin** [comment]
Well im not sure since it checks scenery as well as surface tiles. So in theory you could just cover it in a roof and it would be classed as underground. I'm not really fussed though if you think underground sounds better go for that.

**Comment by IntelOrca** [comment]
I would say sheltered, but that would only make sense if it was being used for rain checks etc.?

**Comment by duncanspumpkin** [comment]
Hmm maybe your right maybe it is used for rain checks as well.

**Comment by Gymnasiast** [comment]
Any ride that's covered by land or scenery for at least 40% is classed as indoor and the guests will ride it when it's raining.

**Comment by duncanspumpkin** [comment]
I'm not sure if its a bug or not but it would appear rides need at least two blocks on top of them to be classed as sheltered. In the vehicle branch i've settled for the following:
```
	sint32 sheltered_length;		// 0x118
	uint8 pad_11C[0x02];
	uint8 no_sheltered_sections;	// 0x11E (?abY YYYY)
```
It would appear that var_11C is never set? I searched for all references to its memory address and there is only two mentions of it and that is reseting it and this function that factors it into the ride calc.

**Comment by kevinburke** [comment]
Interesting. So should I kill the var_11C bits from the function?

**Comment by kevinburke** [comment]
Ok I updated the names here per the suggestion.

**Pull request merged**